Nil pointer dereference when using protocol schemes

## Summary

The `thv run` command panics with a segmentation fault when using protocol schemes like `npx://`, `uvx://`, or `go://` due to improper handling of nil `imageMetadata` in the configuration validation pipeline.

## Steps to Reproduce

1. Run any protocol scheme with the `thv run` command:
   ```bash
   ./bin/thv run --name gitlab npx://@zereight/mcp-gitlab
   ```

2. Observe the warning message about missing provenance information, followed by a panic.

## Expected Behavior

The command should successfully build and run the MCP server from the NPX package without panicking.

## Actual Behavior

```
5:07PM	WARN	MCP server toolhivelocal/npx-zereight-mcp-gitlab:20250821170708 has no provenance information set, skipping image verification
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x40 pc=0x103861f50]
```

## Root Cause Analysis

When using protocol schemes (`npx://`, `uvx://`, `go://`), the system dynamically builds Docker images but does **not** populate `imageMetadata` from the registry since these packages aren't registry entries. This results in `imageMetadata` being `nil` throughout the execution flow.

### Key Areas to Investigate:

- `pkg/runner/config_builder.go` - Lines around metadata field access
- `pkg/runner/env.go` - Environment variable validation with nil metadata  
- Any code that accesses `imageMetadata.*` without first checking `if imageMetadata != nil`

--- a/pkg/registry/types.go
+++ b/pkg/registry/types.go
@@ -219,51 +219,81 @@ type ServerMetadata interface {
 
 // GetName returns the server name
 func (i *ImageMetadata) GetName() string {
+	if i == nil {
+		return ""
+	}
 	return i.Name
 }
 
 // GetDescription returns the server description
 func (i *ImageMetadata) GetDescription() string {
+	if i == nil {
+		return ""
+	}
 	return i.Description
 }
 
 // GetTier returns the server tier
 func (i *ImageMetadata) GetTier() string {
+	if i == nil {
+		return ""
+	}
 	return i.Tier
 }
 
 // GetStatus returns the server status
 func (i *ImageMetadata) GetStatus() string {
+	if i == nil {
+		return ""
+	}
 	return i.Status
 }
 
 // GetTransport returns the server transport
 func (i *ImageMetadata) GetTransport() string {
+	if i == nil {
+		return ""
+	}
 	return i.Transport
 }
 
 // GetTools returns the list of tools provided by the server
 func (i *ImageMetadata) GetTools() []string {
+	if i == nil {
+		return nil
+	}
 	return i.Tools
 }
 
 // GetMetadata returns the server metadata
 func (i *ImageMetadata) GetMetadata() *Metadata {
+	if i == nil {
+		return nil
+	}
 	return i.Metadata
 }
 
 // GetRepositoryURL returns the repository URL
 func (i *ImageMetadata) GetRepositoryURL() string {
+	if i == nil {
+		return ""
+	}
 	return i.RepositoryURL
 }
 
 // GetTags returns the server tags
 func (i *ImageMetadata) GetTags() []string {
+	if i == nil {
+		return nil
+	}
 	return i.Tags
 }
 
 // GetCustomMetadata returns custom metadata
 func (i *ImageMetadata) GetCustomMetadata() map[string]any {
+	if i == nil {
+		return nil
+	}
 	return i.CustomMetadata
 }
 
@@ -274,6 +304,9 @@ func (*ImageMetadata) IsRemote() bool {
 
 // GetEnvVars returns environment variables
 func (i *ImageMetadata) GetEnvVars() []*EnvVar {
+	if i == nil {
+		return nil
+	}
 	return i.EnvVars
 }
 
@@ -281,51 +314,81 @@ func (i *ImageMetadata) GetEnvVars() []*EnvVar {
 
 // GetName returns the server name
 func (r *RemoteServerMetadata) GetName() string {
+	if r == nil {
+		return ""
+	}
 	return r.Name
 }
 
 // GetDescription returns the server description
 func (r *RemoteServerMetadata) GetDescription() string {
+	if r == nil {
+		return ""
+	}
 	return r.Description
 }
 
 // GetTier returns the server tier
 func (r *RemoteServerMetadata) GetTier() string {
+	if r == nil {
+		return ""
+	}
 	return r.Tier
 }
 
 // GetStatus returns the server status
 func (r *RemoteServerMetadata) GetStatus() string {
+	if r == nil {
+		return ""
+	}
 	return r.Status
 }
 
 // GetTransport returns the server transport
 func (r *RemoteServerMetadata) GetTransport() string {
+	if r == nil {
+		return ""
+	}
 	return r.Transport
 }
 
 // GetTools returns the list of tools provided by the server
 func (r *RemoteServerMetadata) GetTools() []string {
+	if r == nil {
+		return nil
+	}
 	return r.Tools
 }
 
 // GetMetadata returns the server metadata
 func (r *RemoteServerMetadata) GetMetadata() *Metadata {
+	if r == nil {
+		return nil
+	}
 	return r.Metadata
 }
 
 // GetRepositoryURL returns the repository URL
 func (r *RemoteServerMetadata) GetRepositoryURL() string {
+	if r == nil {
+		return ""
+	}
 	return r.RepositoryURL
 }
 
 // GetTags returns the server tags
 func (r *RemoteServerMetadata) GetTags() []string {
+	if r == nil {
+		return nil
+	}
 	return r.Tags
 }
 
 // GetCustomMetadata returns custom metadata
 func (r *RemoteServerMetadata) GetCustomMetadata() map[string]any {
+	if r == nil {
+		return nil
+	}
 	return r.CustomMetadata
 }
 
@@ -336,11 +399,17 @@ func (*RemoteServerMetadata) IsRemote() bool {
 
 // GetEnvVars returns environment variables
 func (r *RemoteServerMetadata) GetEnvVars() []*EnvVar {
+	if r == nil {
+		return nil
+	}
 	return r.EnvVars
 }
 
 // GetRawImplementation returns the underlying RemoteServerMetadata pointer
 func (r *RemoteServerMetadata) GetRawImplementation() any {
+	if r == nil {
+		return nil
+	}
 	return r
 }
 

